File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.size();
        string answer;

        // odd substring

        for(int i=0;i<n;i++){
            string temp;
            int l = i-1;
            int r = i+1;

            while(l>=0 && r<n && s[l]==s[r]){
                l--;
                r++;
            }
            temp = s.substr(l+1, r-l-1);
            if(answer.size()<temp.size())answer = temp;
        }    

        // even substring

        for(int i=0;i<n;i++){
            string temp;
            int l = i;
            int r = i+1;

            while(l>=0 && r<n && s[l] == s[r]){
                l--;
                r++;
            }
            temp = s.substr(l+1, r-l-1);
            if(answer.size()<temp.size())answer = temp;
        }

        return answer;
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int n = s.length();
        int window = n;
        for(int i=0; i<n; i++)
        {
            for(int j=0; j<=n-window; j++)
            {
                String temp = s.substring(j, j+window);
                temp = checkPalindrome(temp);
                if(!temp.isEmpty())
                {
                    return temp;
                }
            }
            window--;
        }
        return "";
    }

    public String checkPalindrome(String str)
    {
        for(int i=0, j=str.length()-1; i<str.length() && j>=0; i++, j--)
        {
            if(str.charAt(i)!=str.charAt(j))
            {
                return "";
            }
        }
        return str;
    }
}